grafana/grafana
# Prometheus: Fix some of the alerting queries that use reduce/math operation

**What this PR does / why we need it**:
In https://github.com/grafana/grafana/pull/43622 we have fixed filling out of missing values with null. However, this affected alerting and customers as they build upon the old behaviour, were missing datapoints in prometheus would not be replaced by null. The new behaviour is a bit problematic for now, as any reduce/math operation will return NaN if any null value was passed as part of the input, resulting in triggering an alert. There is a PR that is supposed to introduce Mode to drop [NaN/Inf/Null in Reduction operations]( https://github.com/grafana/grafana/pull/43583) that would fix this issue if users have set that mode, but for now we want to make sure that we are not breaking alerting and we are adding old-way of processing matrix results without filling out nulls for alerting. 

How to test: 
1. Run tns-demo prometheus
2. Create alerting query such as `go_gc_duration_seconds{job="monitoring/node_exporter"}` and step 5s
3. Turn off prom/node-exporter docker image for minute
4. Turn it on
5. Run a query, there shouldn't be a gap and all points should be connected in graph
